library-best-practices/JLBP-14.md

```diff
@@ -5,12 +5,26 @@
   single version for any particular dependency, builds at different points
   in time can see different versions of that dependency (as your dependency
   releases new versions), which can break your library unexpectedly.
-  - Version ranges can even pull in incomplete releases that break your
+
+  - Version ranges cause builds to be non-reproducible. Builds of the
+    same code with the same compiler on different machines at different
+    times can produce different artifacts with arbitrarily different behavior.
+
+  - Version ranges can pull in incomplete releases that break your
     library's build (example:
     https://github.com/GoogleCloudPlatform/appengine-gcs-client/issues/71).
-  - Thus, always specify a single version instead of a version range for
-    dependencies.
+
+  - Version ranges can be a security hole. When a new version of the dependency
+    is pushed to Maven Central, it will be picked up by downstream dependencies
+    automatically. This can be used a by a malicious actor with release privileges
+    to slip new malicious code into projects without proper review. [A version
+    of this attack has been used in the node.js ecosystem to steal 
+    Bitcoins](https://www.theregister.co.uk/2018/11/26/npm_repo_bitcoin_stealer/). 
+
+  Thus, always specify a single version instead of a version range for dependencies.
```

Comment:
This was deliberate. I thought it should be at the same level as the outer list item rather than the inner list item. That is, the sublist are reasons and this is the conclusion. Or does markdown not support this?

Reply:
If you want it at the same level as the outer list item, then do a dash with no spaces before it.

Reply:
That turns it into a new bullet point. What I'm aiming for is 

```
<ul>
  <li>Version ranges cause...
    <ul>
      <li></li>
     <li></li>
    <li></li>
   </ul>
  Thus, always specify ...</li>
</ul>
```

Reply:
Oh ok, you wanted it to be formatted like that. I suppose that's not a big deal - carry on!